File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/ThuongHieuColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ThuongHieuRow } from "./ThuongHieuRow";

export interface ThuongHieuColumns {
    Id: Column<ThuongHieuRow>;
    TenThuongHieu: Column<ThuongHieuRow>;
    Slug: Column<ThuongHieuRow>;
    Logo: Column<ThuongHieuRow>;
    MoTa: Column<ThuongHieuRow>;
    ThuTu: Column<ThuongHieuRow>;
    HienThi: Column<ThuongHieuRow>;
    NoiBat: Column<ThuongHieuRow>;
    CreatedAt: Column<ThuongHieuRow>;
    UpdatedAt: Column<ThuongHieuRow>;
}

export class ThuongHieuColumns extends ColumnsBase<ThuongHieuRow> {
    static readonly columnsKey = 'SanPhamDB.ThuongHieu';
    static readonly Fields = fieldsProxy<ThuongHieuColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/ThuongHieu/ThuongHieuColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.ThuongHieu")]
[BasedOnRow(typeof(ThuongHieuRow), CheckNames = true)]
public class ThuongHieuColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string TenThuongHieu { get; set; }
    public string Slug { get; set; }
    public string Logo { get; set; }
    public string MoTa { get; set; }
    public int ThuTu { get; set; }
    public bool HienThi { get; set; }
    public bool NoiBat { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}